crypto/asn1/ameth_lib.c

```diff
@@ -7,6 +7,11 @@
  * https://www.openssl.org/source/license.html
  */
 
+/* We need to use some engine deprecated APIs */
+#ifndef OPENSSL_NO_DEPRECATED_3_0
+# define OPENSSL_SUPPRESS_DEPRECATED
+#endif
```

Comment:
I don't think you need to wrap this define.  When `OPENSSL_NO_DEPRECATED_3_0` is defined, then `OPENSSL_NO_ENGINE` is defined as well, is it not?

Reply:
I started wrapping these, then decided not to.  This must have slipped through.